react/features/polls/subscriber.js

```diff
@@ -0,0 +1,112 @@
+// @flow
+
+import { getCurrentConference } from '../base/conference';
+import { StateListenerRegistry } from '../base/redux';
+import {
+    NOTIFICATION_TIMEOUT,
+    NOTIFICATION_TYPE,
+    showNotification
+} from '../notifications';
+
+import { receiveAnswer, receivePoll } from './actions';
+import { COMMAND_NEW_POLL, COMMAND_ANSWER_POLL, COMMAND_OLD_POLLS } from './constants';
+import type { Answer, Poll } from './types';
+
+
+const parsePollData = (pollData): Poll | null => {
+    if (typeof pollData !== 'object' || pollData === null) {
+        return null;
+    }
+    const { id, senderId, senderName, question, answers } = pollData;
+
+    if (typeof id !== 'string' || typeof senderId !== 'string' || typeof senderName !== 'string'
+        || typeof question !== 'string' || !(answers instanceof Array)) {
+        return null;
+    }
+
+    const answers2 = [];
+
+    for (const answer of answers) {
+        const voters = new Map();
+
+        for (const [ voterId, voter ] of Object.entries(answer.voters)) {
+            if (typeof voter !== 'string') {
+                return null;
+            }
+            voters.set(voterId, voter);
+        }
+
+        answers2.push({
+            name: answer.name,
+            voters
+        });
+    }
+
+    return {
+        senderId,
+        senderName,
+        question,
+        showResults: true,
+        lastVote: null,
+        answers: answers2
+    };
+};
+
+StateListenerRegistry.register(
+    state => getCurrentConference(state),
+    (conference, store, previousConference) => {
+        if (conference && conference !== previousConference) {
+            conference.room.addListener('xmmp.json_message_received', (senderJid, data) => {
```

Comment:
possible typo: `xmmp.`

Reply:
That constant does not exist. The JSON_MESSAGE_RECEIVED event is internal to LJM, so the constant associated to it is not exported.

More importantly, now that [this PR on LJM](https://github.com/jitsi/lib-jitsi-meet/pull/1671) has been merged, this line will have to change anyway to use the new events (see [this PR on our fork](https://github.com/jade-guiton/jitsi-meet/pull/27)).

Reply:
The change in question was just merged into the PR branch. You can look at 5fc106b to see what changed regarding this issue.